### PR TITLE
RS-387 Attach Resume

### PR DESCRIPTION
**Pull request by l-payan**
- Add MicrosoftAuthClient
- Add MicrosoftGraphClient
- Add System User Model
- Add task to create System User
- Add Upload file to sharepoint interactor
- Upload file in create/update endpoints for referrals
- Add endpoint to download files from sharepoint

### CODE REVIEW

File: app/models/system_user.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+class SystemUser < ApplicationRecord
```
Comment (clopezh) [inline app/models/system_user.rb]: We can change the class name for something like ```ReferralSystemEmailUser```

File: config/initializers/00_dotenv.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+Rails.application.configure do
```
Comment (clopezh) [inline config/initializers/00_dotenv.rb:3]: Do these configs should be in a separate file? or Should they live in development, staging, production rb files?

**Review by clopezh — APPROVED** [review]
Technically the code looks good, great job @l-payan!, but I only have two questions (1 suggestion), Could you please take a look?

File: app/models/system_user.rb (hunk 1 above)
Comment (l-payan) [reply app/models/system_user.rb]: Done!

File: config/initializers/00_dotenv.rb (hunk 2 above)
Comment (l-payan) [reply config/initializers/00_dotenv.rb:3]: This config is build based on .env. so, I don't think we need separeted files, only we need to have the correct env variables in each environment

**Review by clopezh — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**